src/tree/updater_gpu_hist.cu

```diff
@@ -405,11 +407,18 @@ class DeviceHistogram {
     dh::safe_cuda(cudaSetDevice(device_id_));
     if (data_.size() >= kStopGrowingSize) {
       // Recycle histogram memory
-      std::pair<int, size_t> old_entry = *nidx_map_.begin();
-      nidx_map_.erase(old_entry.first);
-      dh::safe_cuda(cudaMemsetAsync(data_.data().get() + old_entry.second, 0,
-                                    n_bins_ * sizeof(GradientSumT)));
-      nidx_map_[nidx] = old_entry.second;
+      if (current_size + n_bins_ * sizeof(GradientSumT) < kStopGrowingSize) {
```

Comment:
Is this line correct? current_size refers to the number of elements (GradientSumT::ValueT) used in data correct? I think sizeof(GradientSumT) will return 8 or 16 and it should be only 2?

Reply:
@RAMitchell You are right, thanks for catching it.